sklearn/tests/test_cross_validation.py

```diff
@@ -385,7 +388,10 @@ def test_label_kfold():
 
     # Check that no label is on both sides of the split
     labels = np.asarray(labels, dtype=object)
-    for train, test in cval.LabelKFold(labels, n_folds=n_folds):
+    for train, test in cval.LabelKFold(labels,
+                                       n_folds=n_folds,
+                                       shuffle=shuffle,
+                                       random_state=rng):
         assert_equal(len(np.intersect1d(labels[train], labels[test])), 0)
 
     # Construct the test data
```

Comment:
This rest of this function seems to be independent of the beginning of the test. IMHO it would make the test more readable by having two check functions:

- `check_label_kfold_integer_labels(shuffle)`
- `check_label_kfold_str_labels(shuffle)`